### PR TITLE
Stat balance 9.0c

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Amazing idea I know:

Adds in a new rare-adjacent bluecross spawn that increases max stats cap
Adds allows the horrorable tie to spawn if a hivemind is killed 
Adds in additional support for balancing around stat caps
Adds in more levels of stats

Added in another crayon spell to get a oddity that increases max stats

### CODE REVIEW

File: code/modules/sanity/inspiration_component.dm [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	//perk
 	var/datum/perk/perk
 	//If we after we are used set are stats to 0
-	var/kill_stats = FALSE
+	var/self_destory = FALSE
```
Comment (SingingSpock) [inline code/modules/sanity/inspiration_component.dm:18]:
```suggestion
	var/self_destroy = FALSE
```
Spelling error; it's best to get all of these proper now

File: code/datums/perks/special.dm [hunk 2]
```diff
@@ -54,3 +54,49 @@
 	return ..()
 
 
+///////////////////
+//Bluecross Perks//
+///////////////////
+
+/datum/perk/skill_cap_exspanding
```
Comment (SingingSpock) [inline code/datums/perks/special.dm:61]:
```suggestion
/datum/perk/skill_cap_expanding
```
Another spelling error

File: code/datums/mob_stats.dm [hunk 3]
```diff
@@ -236,11 +246,17 @@
...
 //Unused but might be good for later additions
 /datum/stat/proc/setValue_withcap(value)
-	if(value > STAT_VALUE_MAXIMUM)
-		src.value = STAT_VALUE_MAXIMUM
+	if(value > stat_cap)
+		src.value = stat_cap
 	else
 		src.value = value
 
+/datum/stat/proc/change_stat_cap(amount)
```
Comment (SingingSpock) [inline code/datums/mob_stats.dm:254]:
```suggestion
/datum/stat/proc/add_stat_cap(amount)
```
A proc named "change stat cap" is unclear whether it adds the value or sets a new one.

File: code/datums/mob_stats.dm [hunk 4]
```diff
@@ -68,17 +68,26 @@
 	var/datum/stat/S = stat_list[statName]
 	S.setValue(Value)
 
+/datum/stat_holder/proc/change_Stat_cap(statName, amount)
```
Comment (SingingSpock) [inline code/datums/mob_stats.dm:71]:
```suggestion
/datum/stat_holder/proc/add_Stat_cap(statName, amount)
```
A proc named "change stat cap" is unclear whether it adds the value or sets a new one.

**Review by SingingSpock — CHANGES_REQUESTED** [review]

**Comment by SingingSpock** [comment]
In addition to the issues with specifics, I don't really think that permitting stat cap increase but only though crayon magic is prudent, especially with so low of a cost.

File: code/datums/mob_stats.dm [hunk 5]
```diff
@@ -68,17 +68,26 @@
...
+	var/datum/stat/S = stat_list[statName]
+	S.change_stat_cap(amount)
```
Comment (Trilbyspaceclone) [inline code/datums/mob_stats.dm:73]:
```suggestion
	S.add_stat_cap(amount)
```

File: code/modules/sanity/inspiration_component.dm [hunk 6]
```diff
@@ -28,7 +28,7 @@
 	else
 		return COMPONENT_INCOMPATIBLE
 	if(kill_stats)
-		kill_stats = TRUE
+		self_destory = TRUE
```
Comment (Trilbyspaceclone) [inline code/modules/sanity/inspiration_component.dm:31]:
```suggestion
		self_destroy = TRUE
```

File: code/modules/sanity/inspiration_component.dm [hunk 7]
```diff
@@ -142,7 +142,7 @@
 			var/datum/perk/oddity/OD = GLOB.all_perks[perk]
 			to_chat(user, SPAN_NOTICE("Instinct tells you more about this anomaly: <span style='color:orange'>[OD]. [OD.desc]</span>"))
 
-	if(kill_stats)
+	if(self_destory)
```
Comment (Trilbyspaceclone) [inline code/modules/sanity/inspiration_component.dm:145]:
```suggestion
	if(self_destroy)
```

File: code/__DEFINES/perks.dm [hunk 8]
```diff
@@ -231,6 +231,9 @@
 
 //Task Perks
 #define PERK_FORCEFUL_REJECTION /datum/perk/forceful_rejection
+//Bluecross perks
+#define PERK_SKILL_CAP_ADDITION /datum/perk/skill_cap_addition
+#define PERK_SKILL_CAP_EXSPANDING /datum/perk/skill_cap_exspanding
```
Comment (Trilbyspaceclone) [inline code/__DEFINES/perks.dm:236]:
```suggestion
#define PERK_SKILL_CAP_EXPANDING /datum/perk/skill_cap_expanding
```

File: code/game/objects/items/oddities.dm [hunk 9]
```diff
@@ -483,6 +483,35 @@
...
+	min_stats = 16
+	//Its 1 then done so we give quite a bit
+	oddity_stats = list(
+		STAT_COG = 18,
+		STAT_VIG = 18
+	)
+	price_tag = 2500 //Bluecross spawn
+	prob_perk = 0
+	kill_stats = TRUE
+	perk = PERK_SKILL_CAP_EXSPANDING
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/oddities.dm:499]:
```suggestion
	perk = PERK_SKILL_CAP_EXPANDING
```

File: code/modules/sanity/sanity_mob.dm [hunk 10]
```diff
@@ -353,7 +353,7 @@ GLOBAL_VAR_INIT(GLOBAL_INSIGHT_MOD, 1)
 					if(owner.stats.addPerk(I.perk))
 						I.perk = null
 
-				if(I.kill_stats)
+				if(I.self_destory)
```
Comment (Trilbyspaceclone) [inline code/modules/sanity/sanity_mob.dm:356]:
```suggestion
				if(I.self_destroy)
```

File: code/datums/perks/special.dm [hunk 11]
```diff
@@ -54,3 +54,49 @@
...
+	active = FALSE
+	passivePerk = FALSE
+
+/datum/perk/skill_cap_exspanding/assign(mob/living/carbon/human/H)
+	..()
+	for(var/stat in ALL_STATS)
+		var/gather_increase = holder.stats.grab_Stat_cap(stat)
+		gather_increase *= 0.5
+		statis_amount = gather_increase
+		holder.stats.change_Stat_cap(gather_increase)
```
Comment (Trilbyspaceclone) [inline code/datums/perks/special.dm:77]:
```suggestion
		holder.stats.add_Stat_cap(gather_increase)
```

File: code/datums/perks/special.dm [hunk 12]
```diff
@@ -54,3 +54,49 @@
...
+	gain_text = "Looking into the stars is starting to becoming productive!"
+	lose_text = "The void above is the same as below."
+	icon_state = "void_eye"
+	active = FALSE
+	passivePerk = FALSE
+
+/datum/perk/skill_cap_addition/assign(mob/living/carbon/human/H)
+	..()
+	for(var/stat in ALL_STATS)
+		holder.stats.change_Stat_cap(30)
```
Comment (Trilbyspaceclone) [inline code/datums/perks/special.dm:96]:
```suggestion
		holder.stats.add_Stat_cap(30)
```

File: code/datums/perks/special.dm [hunk 13]
```diff
@@ -54,3 +54,49 @@
...
+
+/datum/perk/skill_cap_addition/remove()
+	for(var/stat in ALL_STATS)
+		holder.stats.change_Stat_cap(-30)
```
Comment (Trilbyspaceclone) [inline code/datums/perks/special.dm:100]:
```suggestion
		holder.stats.add_Stat_cap(-30)
```

**Pull request merged**